cliquenet/src/lib.rs

```diff
@@ -391,22 +422,27 @@ where
                     }
                 },
                 // One of our connection attempts completed.
-                Some(tt) = self.connect_tasks.join_next() => match tt {
-                    Ok((s, t)) => {
-                        let Some(k) = self.lookup_peer(&t) else {
-                            continue
-                        };
-                        // We only keep the connection if our key is larger than the remote,
-                        // or if we do not have a connection for that key at the moment.
-                        if k < self.key || !self.active.contains_key(&k) {
-                            self.spawn_io(k, s, t)
-                        } else {
-                            warn!(node = %self.key, peer = %k, "dropping new connection");
+                Some(tt) = self.connect_tasks.join_next_with_id() => {
+                    match tt {
+                        Ok((id, (s, t))) => {
+                            let Some(k) = self.lookup_peer(&t) else {
+                                self.task2key.remove(&id);
+                                continue
+                            };
+                            self.on_connect_task_end(id);
+                            // We only keep the connection if our key is larger than the remote,
+                            // or if we do not have a connection for that key at the moment.
+                            if k < self.key || !self.active.contains_key(&k) {
+                                self.spawn_io(k, s, t)
+                            } else {
+                                warn!(node = %self.key, peer = %k, "dropping new connection");
```

Comment:
Is this a warning? Or is it benign when this happens?

Reply:
Agreed, it does not have to be a warning.